### PR TITLE
ante: update nonce check

### LINKED ISSUE #544: Reject transaction sending a lot of transactions in a short period of time
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]

Ethermint: tags `v0.2.0`
Ubuntu 18.04 64bit

__Steps to reproduce:__

1. `make install`
2. `./init.sh`
3. `ethermintcli rest-server --laddr "tcp://localhost:8545" --unlock-key mykey --chain-id "ethermint-1"`
4. run below JavaScript to send transactions(customize `LOOP`, `ADDR_FROM`, `IPADDR`).

```javascript
/////////////////////
// send txs
const LOOP=50;
// ethermintcli q auth account $(ethermintcli keys show mykey -a) -o json | jq -r .value.eth_address
const ADDR_FROM="0xAc7eBb5efF7c508F44F846a8C9C2Eb9dC796CCfB";
// ethermintcli --laddr
const IPADDR='http://localhost:8545';
/////////////////////


const Web3 = require('web3');
const web3 = new Web3();
web3.setProvider(new web3.providers.HttpProvider(IPADDR));

const fn = async() => {
    let nonce = await web3.eth.getTransactionCount(ADDR_FROM);
    console.log("nonce=" + nonce);
    for (let loop=0; loop < LOOP; loop++) {
        const tx = {from: ADDR_FROM, to:"0x00112233445566778899aabbccddeeff00112233", value:"100", gasPrice:"1", nonce: nonce};
        web3.eth.sendTransaction(tx, function(err, hash) {
            if (!err) {
                console.log("hash=" + hash);
            } else {
                console.log("err=" + err);
            }
        })
        .on('receipt', function(receipt){
            console.log('receipt=' + JSON.stringify(receipt.transactionHash));
        })
        nonce++;
    }
}
fn();
```

__Expected behavior:__ [What you expected to happen]

receive all transaction receipts.

__Actual behavior:__ [What actually happened]

sometime fail receive transaction receipts but not detect error.

ethermintd log:

```
[2020-09-25|16:22:40.595] Rejected bad transaction                     module=mempool tx=4E8D7D41A3769F60DCA5388D6E56659F6783C1C5311686EA73621AC294C9BEB4 peerID= res="&{CheckTx:code:3 log:\"invalid sequence: invalid nonce; got 94, expected 90\" gas_used:75892 codespace:\"sdk\" }" err=null
```

__Additional info:__ [Include gist of relevant config, logs, etc.]

Ethermint v0.1.0 too.

If the value of `LOOP` is small(1~3), it is often successful.

### PR DESCRIPTION
**Pull request by noot**
Closes: #544

## Description

- update nonce check in ante handler to reject nonces that are less than the current nonce
______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/ante/eth.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func (nvd NonceVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 	}
 
 	seq := acc.GetSequence()
-	if msgEthTx.Data.AccountNonce != seq {
+	if msgEthTx.Data.AccountNonce < seq {
```
Comment (fedekunze) [inline app/ante/eth.go:259]: can you add a comment to document the change and the reason for it?

**Review by fedekunze — APPROVED** [review]
ACK. Pending bugfix changelog entry too

File: CHANGELOG.md [hunk 2]
```diff
@@ -65,6 +65,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (`x/evm`) [\#496](https://github.com/ChainSafe/ethermint/pull/496) Fix bugs on `journal.revert` and `CommitStateDB.Copy`.
 * (types) [\#480](https://github.com/ChainSafe/ethermint/pull/480) Update [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) coin type to `60` to satisfy [EIP84](https://github.com/ethereum/EIPs/issues/84).
 * (types) [\#513](https://github.com/ChainSafe/ethermint/pull/513) Fix simulated transaction bug that was causing a consensus error by unintentionally affecting the state.
+* (app/ante) [\#550](https://github.com/ChainSafe/ethermint/pull/550) Update ante handler nonce verfication to accept any nonce greater than or equal to the expected nonce to allow to successive transactions.
```
Comment (fedekunze) [inline CHANGELOG.md]: this should go under a new header:

```
## Unreleased

### Bug fixes

* (app/ante) [\#550](https://github.com/ChainSafe/ethermint/pull/550) Update ante handler nonce verification to accept any nonce greater than or equal to the expected nonce to allow to successive transactions. 
```

File: app/ante/eth.go (hunk 1 above)
Comment (fedekunze) [reply app/ante/eth.go:259]: thanks!

File: CHANGELOG.md (hunk 2 above)
Comment (noot) [reply CHANGELOG.md]: fixed!

File: CHANGELOG.md [hunk 3]
```diff
@@ -64,7 +70,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (types) [\#501](https://github.com/ChainSafe/ethermint/pull/501) Fix bech32 encoding error by using the compressed ethereum secp256k1 public key.
 * (`x/evm`) [\#496](https://github.com/ChainSafe/ethermint/pull/496) Fix bugs on `journal.revert` and `CommitStateDB.Copy`.
 * (types) [\#480](https://github.com/ChainSafe/ethermint/pull/480) Update [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) coin type to `60` to satisfy [EIP84](https://github.com/ethereum/EIPs/issues/84).
-* (types) [\#513](https://github.com/ChainSafe/ethermint/pull/513) Fix simulated transaction bug that was causing a consensus error by unintentionally affecting the state.
+* (types) [\#513](https://github.com/ChainSafe/ethermint/pull/513) Fix simulated transaction bug that was causing a consensus error by unintentionally affecting the state.
```
Comment (fedekunze) [inline CHANGELOG.md]:
```suggestion
* (types) [\#513](https://github.com/ChainSafe/ethermint/pull/513) Fix simulated transaction bug that was causing a consensus error by unintentionally affecting the state.
```

**Review by fedekunze — APPROVED** [review]

**Pull request merged**